lib/FusionInventory/Agent/Inventory.pm

```diff
@@ -250,7 +250,26 @@ sub addEntry {
         $entry->{SERIALNUMBER} = $entry->{SERIAL} if !$entry->{SERIALNUMBER}
     }
 
-    push @{$self->{content}{$section}}, $entry;
+    if ($params{identity} && ref $params{identity} eq 'ARRAY') {
+        # check first if a similar entry already exists
+        my @retrievedEntries;
+        while (scalar @retrievedEntries != 1 && scalar $params{identity} > 0) {
+            my $func = shift @{$params{identity}};
+            @retrievedEntries = grep { &$func($entry, $_); } @{$self->{content}{$section}};
+        }
+        if (scalar @retrievedEntries == 1) {
+            # selecting keys that are not already set in entry in inventory
```

Comment:
As you're using callback, you could even set a callback for merging so you can think to define what value becomes mandatory in related module.